p2panda-rs/src/storage_provider/errors.rs

```diff
@@ -114,3 +114,19 @@ pub enum PublishEntryError {
     #[error("Invalid Entry and Operation pair with id {0}")]
     InvalidEntryWithOperation(Hash),
 }
+
+/// `OperationStore` errors.
+#[derive(thiserror::Error, Debug)]
+pub enum OperationStorageError {
+    /// Catch all error which implementers can use for passing their own errors up the chain.
+    #[error("Error occured in OperationStore: {0}")]
+    Custom(String),
+
+    /// A fatal error occured when performing a storage query.
+    #[error("A fatal error occured in OperationStore: {0}")]
+    FatalStorageError(String),
+
+    /// Error which originates in `insert_operation()` when the insertion fails.
```

Comment:
I think this docstring could be rephrased to let implementers know more about the conditions instead of the method name where it is returned, e.g.

> Error returned when insertion of an operation is not possible due to database constraints